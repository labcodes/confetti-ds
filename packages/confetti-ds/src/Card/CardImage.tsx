import React from "react";
export interface CardImageProps {
  /** Specifies the path to the image. */
  src: string;
  /** Specifies an alternate text for the image if the image cannot be displayed for some reason. */
  alt: string;
  /** Sets whether the image has Card's full width. */
  isOverflowed?: boolean;
}

export default function CardImage({
  src,
  alt,
  isOverflowed = false,
}: CardImageProps) {
  return (
    <img
      className={`lab-card-image${
        isOverflowed ? " lab-card-image--overflowed" : ""
      }`}
      src={src}
      alt={alt}
    />
  );
}
