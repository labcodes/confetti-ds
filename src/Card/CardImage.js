import React from "react";
import PropTypes from "prop-types";

export default function CardImage({ src, alt, isOverflowed }) {
  return (
    <img
      className={`lab-card-image${
        isOverflowed ? " lab-card-image--overflowed" : ""
      }`}
      src={src}
      alt={alt}
    />
  );
}

CardImage.propTypes = {
  /** Specifies the path to the image. */
  src: PropTypes.string.isRequired,
  /** Specifies an alternate text for the image if the image cannot be displayed for some reason. */
  alt: PropTypes.string.isRequired,
  /** Sets whether the image has Card's full width. */
  isOverflowed: PropTypes.bool,
};

CardImage.defaultProps = {
  isOverflowed: false,
};
