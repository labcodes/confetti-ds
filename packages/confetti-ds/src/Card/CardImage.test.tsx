import React from "react";
import renderer from "react-test-renderer";

import CardImage from "./CardImage";

describe("CardImage", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<CardImage src="test" alt="test" />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("renders with isOverflowed", async () => {
    const renderedComponent = renderer
      .create(<CardImage src="test" alt="test" isOverflowed />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
