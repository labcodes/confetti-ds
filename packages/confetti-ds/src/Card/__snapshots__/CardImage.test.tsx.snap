// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage renders with base props 1`] = `
<img
  alt="test"
  className="lab-card-image"
  src="test"
/>
`;

exports[`CardImage renders with isOverflowed 1`] = `
<img
  alt="test"
  className="lab-card-image lab-card-image--overflowed"
  src="test"
/>
`;
